server/player/player.go

```diff
@@ -914,6 +915,9 @@ func (p *Player) dropContents() {
 		w.AddEntity(orb)
 	}
 	for _, it := range append(p.inv.Items(), append(p.armour.Items(), p.offHand.Items()...)...) {
+		if _, ok := it.Enchantment(enchantment.Vanishing{}); ok && p.Dead() {
```

Comment:
You can probably just remove the check here for `p.Dead()`, since this is only called on death anyway.

Reply:
Okay ^^